jujugui/static/gui/src/app/app.js

```diff
@@ -989,35 +989,37 @@ YUI.add('juju-gui', function(Y) {
           serviceName: 'jem'
         });
         this.jem = new window.jujulib.environment(this.get('jemUrl'), bakery);
-        this.jem.listEnvironments(function(envList) {
-          // XXX This picks the first environment but we'll want to default to
-          // sandbox mode then allow the user to choose an env.
-          var envData = envList[0];
-          this.set('environmentList', envList);
-          this._renderEnvSwitcher();
-          var doc = window.document;
-          var host = doc.location.hostname;
-          var port = doc.port;
-          var socketUrl = 'wss://' + host;
-          if (port) {
-            socketUrl += ':' + port;
+        this.jem.listEnvironments(function(error, envList) {
```

Comment:
Instead of using `.bind(this)` you can instead use an arrow function here, they are scoped to the context they were defined in.

`this.jem.listEnvironments((error, envList) => { })`